libbeat/publisher/queue/spool/consume.go

```diff
@@ -0,0 +1,118 @@
+package spool
+
+import (
+	"errors"
+	"io"
+
+	"github.com/elastic/beats/libbeat/common/atomic"
+	"github.com/elastic/beats/libbeat/publisher"
+	"github.com/elastic/beats/libbeat/publisher/queue"
+)
+
+type consumer struct {
+	ctx    *spoolCtx
+	closed atomic.Bool
+	done   chan struct{}
+
+	resp chan getResponse
+	requ chan getRequest
+}
+
+type batch struct {
+	events []publisher.Event
+	state  ackState
+	ack    chan batchAckMsg
+}
+
+type ackState uint8
+
+const (
+	batchActive ackState = iota
+	batchACK
+)
+
+func newConsumer(ctx *spoolCtx, requ chan getRequest) *consumer {
+	return &consumer{
+		ctx:    ctx,
+		closed: atomic.MakeBool(false),
+		done:   make(chan struct{}),
+
+		// internal API
+		resp: make(chan getResponse),
+		requ: requ,
+	}
+}
+
+func (c *consumer) Get(sz int) (queue.Batch, error) {
+	log := c.ctx.logger
+
+	if c.closed.Load() || c.ctx.Closed() {
```

Comment:
It would be nice to wrap `c.closed.Load` into a function named `c.Closed`. Just like it is done in `spoolCtx`.